app/controllers/foreman_virt_who_configure/api/v2/configs_controller.rb

```diff
@@ -0,0 +1,120 @@
+
+        before_action :find_resource, :only => %w{show deploy_script update destroy}
+
+        api :GET, '/configs', N_("List of virt-who configurations")
+        api :GET, "/organizations/:organization_id/configs", N_("List of virt-who configurations per organization")
+        param_group :taxonomy_scope, ::Api::V2::BaseController
+        param_group :search_and_pagination, ::Api::V2::BaseController
+
+        def index
+          @configs = resource_scope_for_index
+        end
+
+        api :GET, "/configs/:id/", N_("Show a virt-who configuration")
+        param :id, :identifier, :required => true
+
+        def show
+        end
+
+        api :GET, "/configs/:id/deploy_script/", N_("Renders a deploy script for the specified virt-who configuration")
+        param :id, :identifier, :required => true
+        def deploy_script
+          respond_to do |format|
+            logger.fatal.inspect
+            format.text { render :text => @config.virt_who_config_script }
+            format.sh { render :text => @config.virt_who_config_script }
+            format.json
+          end
+        end
+
+        def_param_group :config do
+          param :foreman_virt_who_configure_config, Hash, :required => true, :action_aware => true do
+            param :name, String, :desc => N_("Configuration name"), :required => true
+            param :interval, Integer, :desc => N_("Configuration interval in minutes"), :required => true
+            param :listing_mode, Integer, :desc => N_("Hypervisor filtering mode, %{unlimited} means no filtering, %{whitelist} means whitelist, %{blacklist} means blacklist") % { :unlimited => Config::UNLIMITED, :whitelist => Config::WHITELIST, :blacklist => Config::BLACKLIST }, :required => false
+            param :whitelist, String, :desc => N_("Hypervisor whitelist"), :required => false
+            param :blacklist, String, :desc => N_("Hypervisor blacklist"), :required => false
+            param :hypervisor_id, String, :desc => N_("Hypervisor identification, one of %s") % Config::HYPERVISOR_IDS.join(', '), :required => true
+            param :hypervisor_type, String, :desc => N_("Hypervisor type, one of %s") % Config::HYPERVISOR_TYPES.keys.join(', '), :required => true
+            param :hypervisor_server, String, :desc => N_("Hypervisor server"), :required => true
+            param :hypervisor_username, String, :desc => N_("Hypervisor username"), :required => true
+            param :hypervisor_password, String, :desc => N_("Hypervisor password"), :required => true
+            param :satellite_url, String, :desc => N_("Satellite server FQDN"), :required => true
+            param :debug, :bool, :desc => N_("Enable debugging output")
+            param :proxy, String, :desc => N_('HTTP Proxy')
+            param :no_proxy, String, :desc => N_("Ignore proxy")
+            param :organization_id, Integer, :required => false, :desc => N_("Scope by organizations") if ::SETTINGS[:organizations_enabled]
```

Comment:
Org id is documented as optional but the API actually returns error when it's not sent.